snuba_sdk/visitors.py

```diff
@@ -153,8 +151,16 @@ def _visit_debug(self, debug: Debug) -> TVisited:
 
 
 class Translation(ExpressionVisitor[str]):
+    def __init__(self, use_entity_aliases: bool = False):
+        # Eventually JOINs will set this to True, but single entity/sub queries
+        # don't support entity aliases.
+        self.use_entity_aliases = use_entity_aliases
```

Comment:
This is probably better addressed in the Join PR (https://github.com/getsentry/snuba-sdk/pull/35).

> Isn't this going to miss the validation rule that all columns in the joins need to reference a table alias?

No in the Join PR I add explicit validation for that.

> This way we should be able to validate different types of queries cleanly and without letting a simple query get into a state that is undefined for it or the join query to get into an invalid state.

I'm not sure how this would prevent that now. Queries in general (not just Join queries) can already exist in an invalid state, since they can be built up over time. They are not expected to be valid until `query.validate()` is called. I did investigate doing the split you are talking about, but ultimately the difference in validation ended up being just a few extra lines of code.

Reply:
Sorry I missed the gap between this file and the previous one. I thought this was the query translator while it was the expression one. Yeah my suggestion would not help with the validation rule for joins.

Reply:
> I did investigate doing the split you are talking about, but ultimately the difference in validation ended up being just a few extra lines of code.

Ok.